### PR TITLE
Extend FS switches support, handle lowerLayerDown ifOperStatus

### LINKED ISSUE #10866: FiberStore S3900-24T4S Switch
**Please do not post bugs or ask questions**

**For help and support you can use our [Discord server](https://t.libren.ms/discord) or our [community site](https://community.librenms.org).**

Please confirm each of the sections below by putting an `x` in the box like `[x]`.

- [x] Please provide **ALL** info asked for [here](http://docs.librenms.org/Support/FAQ/#faq20).
Discovery: https://p.libren.ms/view/9a9500be
Poller: https://p.libren.ms/view/2068a2ea
Snmpwalk: https://github.com/librenms/librenms/files/3877744/S3900_snmpwalk.zip
MIB file: https://img-en.fs.com/file/user_manual/fs-s3900-series-switches-snmp-mib-files.zip
- [x] Please provide as much detail as possible.
The switch is currently detected as "Generic Device" without OS information. The link status of the ports is not shown ( all in blue regardless of the links are up or down). Is it possible to extend the support to cover this model? 
- [x] Please do NOT post more than 10 lines of debug information here, use a pastebin service or GitHub Gists.

Please keep in mind that LibreNMS is a community project, it is run by people like you using their free time.
Which means it may take time before support is added to LibreNMS.

We encourage you to do it yourself and submit it in GitHub as a Pull request. It is not so difficult, most of
it can be done without writing a single line of code. You can find the documentation below :
👉  https://docs.librenms.org/Developing/os/Initial-Detection/

Love librenms? Please consider supporting our collective:
👉  https://t.libren.ms/donations

### PR DESCRIPTION
**Pull request by PipoCanaja**
FS S3900 switches

Also adds a few changes to handle ifOperStatus lowerLayerDown as down. (basically, replace == 'down' tests with != 'up').

fixes #10866 

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Util/Url.php [hunk 1]
```diff
@@ -362,7 +362,7 @@ public static function portLinkDisplayClass($port)
             return "interface-admindown";
         }
 
-        if ($port->ifAdminStatus == "up" && $port->ifOperStatus == "down") {
+        if ($port->ifAdminStatus == "up" && $port->ifOperStatus != "up") {
```
Comment (PipoCanaja) [inline LibreNMS/Util/Url.php]: Can now handle stats like lowerLayerDown

File: includes/caches/ports.inc.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
     $data['up'] = array('query' => "SELECT COUNT(*) FROM ports AS I, devices AS D WHERE I.`deleted` = '0' AND D.`device_id` = I.`device_id` AND I.`ignore` = '0' AND D.`ignore` = '0' AND I.`ifOperStatus` = 'up'",);
 
-    $data['down'] = array('query' => "SELECT COUNT(*) FROM ports AS I, devices AS D WHERE I.`deleted` = '0' AND D.`device_id` = I.`device_id` AND I.`ignore` = '0' AND D.`ignore` = '0' AND I.`ifOperStatus` = 'down' AND I.`ifAdminStatus` = 'up'");
+    $data['down'] = array('query' => "SELECT COUNT(*) FROM ports AS I, devices AS D WHERE I.`deleted` = '0' AND D.`device_id` = I.`device_id` AND I.`ignore` = '0' AND D.`ignore` = '0' AND I.`ifOperStatus` != 'up' AND I.`ifAdminStatus` = 'up'");
```
Comment (PipoCanaja) [inline includes/caches/ports.inc.php]: Can now handle state like lowerLayerDown

File: mibs/fs/GBNDeviceStack-MIB [hunk 3]
```diff
@@ -15,14 +15,14 @@
         ;
         
     gbnDeviceStackMib MODULE-IDENTITY
-        LAST-UPDATED    "2013/07/09"  -- July 09, 2013
+        LAST-UPDATED    "201307090000Z" -- July 09, 2013
```
Comment (PipoCanaja) [inline mibs/fs/GBNDeviceStack-MIB]: correct the mib to get it compile ...

**Comment by Jellyfrog** [comment]
There are a lot more places that checks `ifOperStatus`, for example Port model, `scopeIsDown`-function.

**Comment by PipoCanaja** [comment]
> There are a lot more places that checks `ifOperStatus`, for example Port model, `scopeIsDown`-function.

Thx, this one is patched. Will have to do a bunch of 'greps' I guess to find all ...

**Comment by PipoCanaja** [comment]
Should be all better now, after a "grep -r ifOperStatus". At least, ready for review

**Comment by PipoCanaja** [comment]
@librenms/reviewers Ready for review. On the way to get below 100 issues :D Don't know if it is doable by end of 2019 :D

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-60-release-changelog-january-2020/10863/1

**Pull request merged**